vpn/data.go

```diff
@@ -259,31 +271,41 @@ func (d *data) SetupKeys(dck *dataChannelKey) error {
 // done together in the same function.
 // TODO accept state for symmetry
 func (d *data) EncryptAndEncodePayload(plaintext []byte, dcs *dataChannelState) ([]byte, error) {
-	blockSize := dcs.dataCipher.blockSize()
+	if len(plaintext) == 0 {
+		return []byte{}, fmt.Errorf("%w: nothing to encrypt", errCannotEncrypt)
+	}
+	if dcs == nil || dcs.dataCipher == nil {
+		return []byte{}, fmt.Errorf("%w: %s", errCannotEncrypt, fmt.Errorf("data chan not initialized"))
```

Comment:
* [x] MIV-01-003 Possible DoS via nil Pointer Dereference

👍 for also adding regress tests for it